Simplenote/src/androidTest/java/com/automattic/simplenote/smoke/test/TestRunner.java

```diff
@@ -0,0 +1,405 @@
+package com.automattic.simplenote.smoke.test;
+
+import androidx.test.rule.ActivityTestRule;
+
+import com.automattic.simplenote.NotesActivity;
+import com.automattic.simplenote.smoke.data.DataProvider;
+import com.automattic.simplenote.smoke.data.NoteDTO;
+import com.automattic.simplenote.smoke.data.SignUpDataProvider;
+import com.automattic.simplenote.smoke.pages.IntroPage;
+import com.automattic.simplenote.smoke.pages.LoginPage;
+import com.automattic.simplenote.smoke.pages.MainPage;
+import com.automattic.simplenote.smoke.pages.SettingsPage.SortOrder;
+import com.automattic.simplenote.smoke.utils.TestUtils;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class TestRunner {
+    @Rule
+    public ActivityTestRule<NotesActivity> mActivityTestRule = new ActivityTestRule<>(NotesActivity.class, true, true);
+
+    private static String email;
+    private static String password;
+
+    @Before
+    public void init() {
+        TestUtils.logoutIfNecessary(DataProvider.UNKNOWN_EMAIL);
+
+        if (email == null || password == null) {
+
+            email = SignUpDataProvider.generateEmail();
+            password = SignUpDataProvider.generatePassword();
+
+            System.out.println("Email: " + email);
+            System.out.println("Password: " + password);
+
+            new IntroPage()
+                    .openSignUp()
+                    .signUp(email, password)
+                    .logout(email);
+        }
+    }
+
+    @After
+    public void tearDown() {
+    }
+
+    @Test
+    public void testLogin() {
+
+        new IntroPage()
+                .goToLoginWithEmail()
+                .login(email, password);
+
+        new MainPage()
+                .isOpened()
+                .logout(email);
+    }
+
+    @Test
+    public void testLogout() {
+
+        if (!TestUtils.logoutIfNecessary(email)) {
+            new IntroPage()
+                    .goToLoginWithEmail()
+                    .login(email, password);
+
+            new MainPage()
+                    .isOpened()
+                    .logout(email);
+        }
+    }
```

Comment:
What's the benefit of having this as a separate test from `testLogin()` (given that the login test also logs out at the end)?

Reply:
Running all tests separately without dependencies on between them is an approach. For example, if we use a test management tool and implement test scenarios, we should separate them like this. All test scenarios should be mapped at least one test automation script. But if this is not necessary we can go with merging them and cut off the relationship between scenarios and scripts. 

Yes, `testLogin()` currently has the `logout` part. But that is just a postcondition. Somehow we may want to run `testLogin()` script with removing `logout` part.

All these are about the approach. We can go with any of this method depend on the requirements.